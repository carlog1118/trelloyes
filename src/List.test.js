import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe ('List Component', () => {
    const props = {
        header: 'the header',
        cards: [
          {
            id: 'id1',
            title: 'title 1',
            content: 'content 1'
          },
          {
            id: 'id2',
            title: 'title 2',
            content: 'content 2'
          }
        ]
      };
    it ('Renders without crashing', () => {
        const div= document.createElement('div');
        ReactDOM.render(<List{...props}/>,  div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it ('Renders in UI as expected', () => {
        const tree= renderer
            .create(<List{...props}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});