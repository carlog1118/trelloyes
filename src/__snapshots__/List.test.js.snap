// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component Renders in UI as expected 1`] = `
<section
  className="List"
>
  <header>
    the header
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        title 1
      </h3>
      <p>
        content 1
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        title 2
      </h3>
      <p>
        content 2
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
